janitor/xarray/functions.py

```diff
@@ -0,0 +1,97 @@
+"""
+Functions to augment XArray DataArrays and Datasets with additional
+functionality.
+"""
+
+
+import xarray as xr
+import numpy as np
+
+from typing import Union
+
+from .utils import register_dataarray_method, register_dataset_method
+
+
+@register_dataarray_method
+def clone_using(
+    da: xr.DataArray,
+    np_arr: np.array,
+    use_coords: bool = True,
+    use_attrs: bool = False,
+    new_name: str = None,
+) -> xr.DataArray:
+    """
+    Given a NumPy array, return an XArray ``DataArray`` which contains the same
+    dimension names and (optionally) coordinates and other properties as the
+    supplied ``DataArray``.
+
+    This is similar to ``xr.DataArray.copy()`` with more specificity for
+    the type of cloning you would like to perform - the different properties
+    that you desire to mirror in the new ``DataArray``.
+
+    If the coordinates from the source ``DataArray`` are not desired, the shape
+    of the source and new NumPy arrays don't need to match.
+    The number of dimensions do, however.
```

Comment:
Could we add a quick usage example here, like those found on pyjanitor's functions? Just to remind users of the API.

Reply:
Will do. Forgot about this for whatever reason.